### PR TITLE
Refactor rmrk/CreateToken

### PR DESCRIPTION
**Pull request by pbkompasz**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring

## Context

- [X] Part of #4750
- [X] Part of #6392 
- [ ] Requires deployment <snek/rubick/worker>

#### Before submitting pull request, please make sure:

- [X] My contribution builds **clean without any errors or warnings**
- [X] I've merged recent default branch -- **main** and I've no conflicts
- [X] I've tried to respect high code quality standards
- [X] I've didn't break any original functionality

#### Optional

- [ ] I've tested it at </ksm/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Did your issue had any of the "$" label on it?

- [X] Fill up your DOT address: [Payout](https://kodadot.xyz/dot/transfer?target=16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt&usdamount=50&donation=true)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [ ] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

## Copilot Summary
copilot:summary

copilot:poem

### CODE REVIEW

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 4.57 USD/DOT ~ 10.941 $DOT
🧗 [16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt](https://polkadot.subscan.io/account/16hqk8Tjugo9AwSzKkxXc1dsRK7iJetpx1YBaoXYwdHzuNAt)
🔗 [0x11fa5720ef9ebe6406cbc46cc0e7c85d86ab42f1e0bdbb53c68c11745643ef4b](https://polkadot.subscan.io/extrinsic/0x11fa5720ef9ebe6406cbc46cc0e7c85d86ab42f1e0bdbb53c68c11745643ef4b)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Comment by prury** [comment]
checking, lets see if it solves #6916

**Comment by pbkompasz** [comment]
@prury Sorry about that, I couldn't test it locally. Can you check if it works now?

**Comment by pbkompasz** [comment]
@prury Sorry about that, I couldn't test it locally. Can you check if it works now?

File: components/rmrk/Create/CreateToken.vue [hunk 1]
```diff
@@ -97,296 +90,297 @@ import { CreatedNFT, Interaction } from '@kodadot1/minimark/v1'
...
+    return
   }
 
-  @Watch('accountId', { immediate: true })
-  hasAccount(value: string, oldVal: string) {
-    if (shouldUpdate(value, oldVal)) {
-      this.fetchCollections()
+  isLoading.value = true
+  status.value = 'loader.ipfs'
+  const { urlPrefix } = usePrefix()
```
Comment (roiLeo) [inline components/rmrk/Create/CreateToken.vue:220]:
```suggestion
```

File: components/rmrk/Create/CreateToken.vue [hunk 2]
```diff
@@ -97,296 +90,297 @@ import { CreatedNFT, Interaction } from '@kodadot1/minimark/v1'
...
+  pageId: string
+  nftName: string
+  toCollectionPage: boolean
+}) => {
+  showNotification(
+    `You will go to the detail in ${DETAIL_TIMEOUT / 1000} seconds`
+  )
+  const subPath = toCollectionPage ? 'collection' : 'gallery'
+  const go = () => {
+    const navigateTo: Location = {
```
Comment (roiLeo) [inline components/rmrk/Create/CreateToken.vue:364]:
```suggestion
    const navigateTo = {
```

File: components/rmrk/Create/CreateToken.vue [hunk 3]
```diff
@@ -97,296 +90,297 @@ import { CreatedNFT, Interaction } from '@kodadot1/minimark/v1'
...
+import CollapseWrapper from '@/components/shared/collapse/CollapseWrapper.vue'
+import Loader from '@/components/shared/Loader.vue'
+import BalanceInput from '@/components/shared/BalanceInput.vue'
+import BaseTokenForm from '@/components/base/BaseTokenForm.vue'
+import BasicSwitch from '@/components/shared/form/BasicSwitch.vue'
+import SubmitButton from '@/components/base/SubmitButton.vue'
+import RoyaltyForm from '@/components/bsx/Create/RoyaltyForm.vue'
+
+import type { BaseTokenType } from '@/components/base/types'
+import type { Location } from 'vue-router/types/router'
```
Comment (roiLeo) [inline components/rmrk/Create/CreateToken.vue:108]:
```suggestion
```

**Review by roiLeo — APPROVED** [review]
code lgtm

**Comment by prury** [comment]
i understand is a refactor, but since this functionality is not working (classic minting on rmrk1), i am not able to go further into testing. 
if you folks want to maintain the refactoring for the sake of nuxt3, lets go. 
if not, lmk. 

wdyt @roiLeo

**Comment by roiLeo** [comment]
> i understand is a refactor, but since this functionality is not working (classic minting on rmrk1

Does it work on rmrkv2? As it use the same component 

> if you folks want to maintain the refactoring for the sake of nuxt3, lets go. if not, lmk.

if it work on beta/canary but not on this preview then we need to fix it.
if it doesn't work for either of them, then we can fix it latter.

**Comment by prury** [comment]
> > i understand is a refactor, but since this functionality is not working (classic minting on rmrk1
> 
> Does it work on rmrkv2? As it use the same component

yes, just minted one there(canary): 
https://kusama.subscan.io/tx/0x62bfe4a6ed827515f6ac483e27c19ec1ae4473eb87681c88546996bdd49a1e44

> 
> > if you folks want to maintain the refactoring for the sake of nuxt3, lets go. if not, lmk.
> 
> if it work on beta/canary but not on this preview then we need to fix it. if it doesn't work for either of them, then we can fix it latter.

was not working already, I've linked the issue few messages above

> 
> edit: more infos [here](https://github.com/kodadot/nft-gallery/issues/7044#issuecomment-1700788734)

fix it later then?

**Comment by yangwao** [comment]
@pbkompasz seems doesn't builds 👀

image

**Review by yangwao — APPROVED** [review]

**Comment by yangwao** [comment]
> fix it later then?

if there is anything left, let's make follow up @prury 

merging till then as it was building, only wasn't test passing but that was not working before this pr iirc

**Pull request merged**